tests/neuro_san/client/abstract_agent_evaluator.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import List
+from typing import Dict
+
+from leaf_common.parsers.dictionary_extractor import DictionaryExtractor
+
+from neuro_san.message_processing.basic_message_processor import BasicMessageProcessor
+
+from tests.neuro_san.client.agent_evaluator import AgentEvaluator
+from tests.neuro_san.client.assert_forwarder import AssertForwarder
+
+
+class AbstractAgentEvaluator(AgentEvaluator):
+    """
+    Abstract AgentEvaluator implementation that does common preparation for tests
+    """
```

Comment:
Added a new ValueAgentEvaluator and that guy had a lot of the same functionality as the original KeywordsAgentEvalautor.  So the common logic comes here.